docs/getting-started/auth.adoc

```diff
@@ -1,15 +1,169 @@
+----
+AuthService authService = mobileCore.getInstance(AuthService.class);
+----
+
+Any subsequent `MobileCore#getInstance` method will return the same instance of `AuthService`.
+
+Before the `AuthService` can be used used `AuthService#init` must be invoked once in an app.
+
+If `AuthService#init` is not invoked then an `IllegalStateException` will be thrown when using the
+service.
+
+[source,java]
+----
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .build();
+
+// Only invoke this once, every subsequent retrieval of the AuthService will retrieve the same, already initialized, instance.
+authService.init(context, authServiceConfig);
+----
+
+=== Retrieving the current user
+
+To retrieve the current authenticated user the `AuthService#currentUser` method can be invoked. This will be `null` if there is
+no user authenticated. So it can be used to check whether to start the authentication flow or not.
+
+[source,java]
+----
+// authService already initialized.
+UserPrincipal currentUser = authService.currentUser();
+
+if (currentUser) {
+    // User is authenticated, get the users name
+    String userName = currentUser.getName();
+    // Get the users email address
+    String userEmail = currentUser.getEmail();
+    // Get the access token of the authenticated user
+    String accessToken = currentUser.getAccessToken();
+} else {
+    // User is not authenticates, start authentication flow
+}
+----
+
+=== Authenticating
+
+To start the authentication invoke the `AuthService#login` method.
+
+[source,java]
+----
+// authService already initialized.
+AuthService authService = mobileCore.getInstance(AuthService.class);
+
+static int LOGIN_RESULT_CODE = 1;
+Activity myActivity = this.getActivity();
+
+// Build the options object and start the authentication flow.
+OIDCAuthenticateOptions options = new OIDCAuthenticateOptions(myActivity, LOGIN_RESULT_CODE);
+
+Callback authCallback = new Callback<UserPrincipal>() {
+    @Override
+    public void onSuccess(UserPrincipal principal) {
+        // User authenticated in, continue on..
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // An error occurred during login.
+    }
+}
+
+authService.login(options, authCallback);
+----
+
+Once the browser returns to the app the result can be handled. In the activity provided to the
+`login` method override `onActivityResult`. This handler should always invoke
+`AuthService#handleAuthResponse`, providing the `Intent`. This will exchange the temporary tokens
+returned from `AuthService#login` for long-life tokens and will provide a
+`UserPrincipal` which can be used to access a users details.
+
+More information about the user returned is available in link:../core/README.adoc[the auth module JavaDocs].
+
+[source,java]
+----
+@Override
+public void onActivityResult(int requestCode, int resultCode, Intent data) {
```

Comment:
When & why would I ever need a different implementation of this method?
What happens if I don't implement it?

Reply:
onActivityResult would handle more than just our single result in a real app. As for why our specific `if` block would be different, apart from logging I doubt it would.

There is an explanation above about what this does. However, I can add more information about what happens if you don't do it.